test/serializer/instant-render/EmptyBuiltInProps.js

```diff
@@ -0,0 +1,19 @@
+// instant render
+// add at runtime:var __empty = {};
+
+function f(c) {
+  var a = {};
+  if (c) {
+    a.foo = 42;
+  }
+
+  return a;
+}
+
+global.__optimize && __optimize(f);
+
+inspect = function() {
+  let result = f(false);
+  if (!global.__optimize) result.foo = {};
+  return JSON.stringify(result);
```

Comment:
Maybe `return !("foo" in result) || result.foo === __empty`. Not sure if the __empty would get rewritten. That case, you `add at runtime:var __empty = {empty: true};`, and then check if `result.foo.empty`.